mon-pix/app/pods/components/module/grain/component.js

```diff
@@ -30,7 +30,9 @@ export default class ModuleGrain extends Component {
           return undefined;
         }
       })
-      .filter((element) => ModuleGrain.AVAILABLE_ELEMENT_TYPES.includes(element.type));
+      .filter((element) => {
```

Comment:
Du coup, il y aura un moment où le front ne sera pas capable de consommer le Didacticiel => page blanche.

Je pense on peut se le permettre cette fois ci mais c'est pas un choix anodin !

Reply:
Pourquoi est-ce que ça finirait en page blanche ?

Reply:
C'est ce qui est arrivé ce matin en mob. Et ça peut arriver en prod car les déploiements ne sont pas instantanés : un front d'une version `n` peut discuter avec une API d'une version `n+1`.

Reply:
Je vois bien l'idée d'un front vn qui discute avec une api vn+1.
Mais du coup concrètement, dans le genre de cas qui est le nôtre ici, avec ce genre de code que tu soulignes, qu'est-ce qui pourrait arriver ?

Reply:
Tu peux constater la page blanche en supprimant cette modif côté front (vn) mais conservant les modifs côté back (vn+1).

Je pense concrètement à un utilisateur qui n'aurait pas rafraichi la page de détails du didacticiel (front en vn) puis qui demanderai à accéder au passage (API renvoie le module vn+1 avec un component stepper) => cette méthode throw => page blanche.

Reply:
You're right c'est de ça dont on s'était rendu compte et qui nous a fait écrire ce test.
Dans ce cas, quelle aurait pu être notre stratégie ?

Reply:
Commencer par faire cette évolution côté front, et la livrer.

Une fois qu'elle arriverait en prod on pourrait merger la partie API de cette PR.

Toujours penser à la rétrocompatibilité ;)

Reply:
ah mais oui, merci pour l'éclaircissement :)